src/syft/core/nodes/abstract/worker.py

```diff
@@ -86,3 +93,60 @@ def _register_services(self) -> None:
                     self.msg_router[handler_type_subclass] = service_instance
 
         self.services_registered = True
+
+    @type_hints
+    def _fetch_network(self) -> None:
+        """
+        This method allow connecting to a main orchestrator that can
+        send the information about the surrounding networks to update
+        the configuration jsons.
+
+        A worker may override this method to allow for caching network
+        information in systems like Redis.
+        Also, extended worker is able to request updated configuration
+        from a remote system.
+        """
+        # read from json and cache network information.
+        with open('network.json') as f:
+          self.network = json.load(f)
+
+    def _learn_my_specs(self) -> None:
+        with open('specs.json') as f:
+          self.specs = json.load(f)
+
+    @type_hints
+    def sign_message(self, msg: SyftMessage) -> SyftMessage:
```

Comment:
So I think that this functionality should be wrapped in a Service instead of put on the worker class. All functionality which processes messages of a certain type on a worker should be packaged into a Service which manages messages of that type.

I also have some services which manage "all messages which are subclasses of X message type"

Reply:
Functionally equivalent - but in previous versions of Syft the "Worker" class has gotten insanely large because we just add a billion methods for all the kinds of methods it processes. Services are more about code organization than anything.